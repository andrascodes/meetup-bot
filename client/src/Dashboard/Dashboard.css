.Dashboard {
  margin-top: 2em;
}

.Chart > div {
  margin: auto;
}

.TotalErrors {
  margin-top: -2em;
}