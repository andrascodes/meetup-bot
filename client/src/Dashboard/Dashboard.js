import React, { Component } from 'react'
import { BarChart, XAxis, YAxis, Bar, ResponsiveContainer, Tooltip, Pie, PieChart } from 'recharts'

import { createServiceApi } from '../lib/serviceApi' 

import './Dashboard.css'

import { 
  Panel as TotalUsers, 
  Panel as ActiveToday,
  Panel as ActiveMonthly,
  Panel as TotalMessages,
  Panel as AvgMessages,
  Panel as AvgLength,
  Panel as TotalErrors
} from './Panel'

export class Dashboard extends Component {
  constructor(props) {
    super(props)

    this.state = {
      users: {
        activeUsersByMonth: [],
        activeUsersLastMonth: 0,
        activeUsersToday: 0,
        activeUsersTwoMonthsAgo: 0,
        activeUsersYesterday: 0,
        totalUsers: 0
      },
      messages: {
        totalMessages: 0,
        avgMessages: 0,
        avgLength: '',
        outgoingMessages: 0,
        deliveredAndReadMessages: 0,
        deliveredNotReadMessages: 0,
        notDeliveredMessages: 0,
        messageErrors: 0
      }
    }
  }
  
  loadDashboard = createServiceApi(fetch, 'http://localhost:5000/api').loadDashboard

  componentDidMount() {
    this.loadDashboard()
    .then(data => 
      this.setState(state => ({
        ...data
      }))
    )
  }

  renderCustomizedLabel = ({ cx, cy, midAngle, innerRadius, outerRadius, percent, index }) => {
    const RADIAN = Math.PI / 180
    const radius = innerRadius + (outerRadius - innerRadius) * 0.5
    const x  = cx + radius * Math.cos(-midAngle * RADIAN)
    const y = cy  + radius * Math.sin(-midAngle * RADIAN)
  
    return (
      <text x={x} y={y} fill="white" textAnchor={x > cx ? 'start' : 'end'} 	dominantBaseline="central">
        {`${(percent * 100).toFixed(0)}%`}
      </text>
    );
  };

  render() {

    return (
      <div className="container-fluid">
        <h1>Dashboard</h1>

        <div className="row Dashboard">
          <div className="UserChart Chart">
            <h3 className="text-center" >Active Users By Month</h3>
            <ResponsiveContainer width="80%" minHeight={360} height="80%" style={{ margin: 'auto' }}>
              <BarChart data={this.state.users.activeUsersByMonth} >
                <XAxis dataKey="month"/>
                <YAxis />
                <Tooltip />
                <Bar dataKey="count" />
              </BarChart>
            </ResponsiveContainer>
          </div>
        </div>

        <div className="row">
          <div className="col-lg-4 col-md-4">
            <TotalUsers 
              icon={'fa-users'}
              number={this.state.users.totalUsers}
              label={`Total Users`}
            />
          </div>
          
          <div className="col-lg-4 col-md-4">
            <ActiveToday 
              icon={'fa-user'}
              number={this.state.users.activeUsersToday}
              previous={this.state.users.activeUsersYesterday}
              label={`Active Users Today`}
            />
          </div>

          <div className="col-lg-4 col-md-4">
            <ActiveMonthly
              icon={'fa-user-circle-o'}
              number={this.state.users.activeUsersLastMonth}
              previous={this.state.users.activeUsersTwoMonthsAgo}
              label={`Active Users This Month`}
            />
          </div>
        </div>

        <div className="row">
          <div className="MessageChart Chart col-lg-12 col-md-12">
            <h3 className="text-center" >Delivered, Read, Undelivered Messages Ratio</h3>
            <ResponsiveContainer width="80%" minHeight={360} height="80%" style={{ margin: 'auto' }}>
              <PieChart width={730} height={250}>
                <Pie 
                  data={[
                    {
                      name: 'Delivered and Read',
                      value: this.state.messages.deliveredAndReadMessages
                    },
                    {
                      name: 'Delivered, but Not Read',
                      value: this.state.messages.deliveredNotReadMessages
                    },
                    {
                      name: 'Not Delivered',
                      value: this.state.messages.notDeliveredMessages
                    },
                  ]} 
                  cx="50%" cy="50%" 
                  outerRadius={125} fill="#31708f"
                  labelLine={false} label={this.renderCustomizedLabel}
                />
                <Tooltip />
              </PieChart>
            </ResponsiveContainer>
          </div>
        </div>
        <div className="row">
            
          <div className="col-lg-4 col-md-4"></div>
          
          <div className="col-lg-4 col-md-4">
            <div className="TotalErrors">
              <TotalErrors 
                icon={'fa-exclamation-triangle'}
                number={this.state.messages.messageErrors}
                label={`No. of message errors`}
              />
            </div>
          </div>

          <div className="col-lg-4 col-md-4"></div>

        </div>

        <div className="row">
          <div className="col-lg-4 col-md-4">
            <TotalMessages 
              icon={'fa-comments'}
              number={this.state.messages.totalMessages}
              label={`Total Messages`}
            />
          </div>

          <div className="col-lg-4 col-md-4">
            <AvgMessages 
              icon={'fa-comment'}
              number={Number(this.state.messages.avgMessages).toFixed(2)}
              label={`Avg. no. messages in a conversation`}
            />
          </div>

          <div className="col-lg-4 col-md-4">
            <AvgLength 
              icon={'fa-commenting'}
              number={this.state.messages.avgLength}
              label={`Avg. length of a conversation`}
            />
          </div>
        </div>
       
      </div>
    )
  }
}