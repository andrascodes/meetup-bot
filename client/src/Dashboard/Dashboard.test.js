import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import { Dashboard } from './Dashboard'

describe(`Dashboard`, () => {
  
  beforeEach(() => {
    global.localStorage = {
      clear: jest.fn()
    }
  })

  test('renders Redirect to /login when state.error.status is 401: Unauthorized', () => {
    const wrapper = mount(
      <MemoryRouter>
        <Dashboard error={{ status: 401 }}/>
      </MemoryRouter>
    )
    expect(global.localStorage.clear).toHaveBeenCalled()
    expect(wrapper.find('Redirect').prop('to')).toEqual('/login')
  })

  test('renders Error when state.error is truthy', () => {
    const wrapper = mount(<Dashboard />)
    wrapper.setState({ error: { status: 500 } })
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  test('renders correctly', () => {
    const wrapper = mount(<Dashboard />)
    expect(toJson(wrapper)).toMatchSnapshot()

    wrapper.setState({ 
      "users": {
        "totalUsers": 1,
        "activeUsersToday": 0,
        "activeUsersYesterday": 0,
        "activeUsersLastMonth": 1,
        "activeUsersTwoMonthsAgo": 0,
        "activeUsersByMonth": [
          {
            "month": "April",
            "count": 1
          }
        ]
      },
      "messages": {
        "totalMessages": 34,
        "avgMessages": 5.666666666666667,
        "avgLength": "00:08",
        "outgoingMessages": 17,
        "deliveredAndReadMessages": 16,
        "deliveredNotReadMessages": 0,
        "notDeliveredMessages": 1,
        "messageErrors": 0
      }
    })
    expect(toJson(wrapper)).toMatchSnapshot()
  })

})