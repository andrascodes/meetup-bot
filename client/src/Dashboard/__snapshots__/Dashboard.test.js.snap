exports[`Dashboard renders Error when state.error is truthy 1`] = `
<Dashboard>
  <Component>
    <div
      className="container-fluid">
      <h2>
        Error. 
        Try again later!
      </h2>
    </div>
  </Component>
</Dashboard>
`;

exports[`Dashboard renders correctly 1`] = `
<Dashboard>
  <div
    className="container-fluid">
    <h1>
      Dashboard
    </h1>
    <div
      className="row Dashboard">
      <div
        className="UserChart Chart">
        <h3
          className="text-center">
          Active Users By Month
        </h3>
        <ResponsiveContainer
          debounce={0}
          height="80%"
          minHeight={360}
          style={
            Object {
              "margin": "auto",
            }
          }
          width="80%">
          <div
            className="recharts-responsive-container"
            style={
              Object {
                "height": "80%",
                "maxHeight": undefined,
                "minHeight": 360,
                "minWidth": undefined,
                "width": "80%",
              }
            }>
            <AnimationDecorator(ComposedDataDecorator(BarChart))
              data={Array []}
              height={0}
              layout="horizontal"
              margin={
                Object {
                  "bottom": 5,
                  "left": 5,
                  "right": 5,
                  "top": 5,
                }
              }
              stackOffset="none"
              width={0} />
            <ResizeDetector
              handleHeight={true}
              handleWidth={true}
              onResize={[Function]}>
              <div
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "visibility": "hidden",
                    "zIndex": -1,
                  }
                }>
                <div
                  onScroll={[Function]}
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "visibility": "hidden",
                      "zIndex": -1,
                    }
                  }>
                  <div
                    style={
                      Object {
                        "height": 10,
                        "left": 0,
                        "position": "absolute",
                        "top": 0,
                        "width": 10,
                      }
                    } />
                </div>
                <div
                  onScroll={[Function]}
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "visibility": "hidden",
                      "zIndex": -1,
                    }
                  }>
                  <div
                    style={
                      Object {
                        "height": "200%",
                        "left": 0,
                        "position": "absolute",
                        "top": 0,
                        "width": "200%",
                      }
                    } />
                </div>
              </div>
            </ResizeDetector>
          </div>
        </ResponsiveContainer>
      </div>
    </div>
    <div
      className="row">
      <div
        className="col-lg-4 col-md-4">
        <Component
          icon="fa-users"
          label="Total Users"
          number={0}>
          <div
            className="panel panel-info">
            <div
              className="panel-heading">
              <div
                className="row">
                <div
                  className="col-xs-3">
                  <i
                    className="fa fa-users fa-5x" />
                </div>
                <div
                  className="col-xs-9 text-right">
                  <div
                    className="huge"
                    style={
                      Object {
                        "fontSize": "40px",
                      }
                    }>
                    0
                  </div>
                  <div>
                    Total Users
                  </div>
                </div>
              </div>
            </div>
          </div>
        </Component>
      </div>
      <div
        className="col-lg-4 col-md-4">
        <Component
          icon="fa-user"
          label="Active Users Today"
          number={0}
          previous={0}>
          <div
            className="panel panel-warning">
            <div
              className="panel-heading">
              <div
                className="row">
                <div
                  className="col-xs-3">
                  <i
                    className="fa fa-user fa-5x" />
                </div>
                <div
                  className="col-xs-9 text-right">
                  <div
                    className="huge"
                    style={
                      Object {
                        "fontSize": "40px",
                      }
                    }>
                    0
                  </div>
                  <div>
                    Active Users Today
                  </div>
                </div>
              </div>
            </div>
          </div>
        </Component>
      </div>
      <div
        className="col-lg-4 col-md-4">
        <Component
          icon="fa-user-circle-o"
          label="Active Users This Month"
          number={0}
          previous={0}>
          <div
            className="panel panel-warning">
            <div
              className="panel-heading">
              <div
                className="row">
                <div
                  className="col-xs-3">
                  <i
                    className="fa fa-user-circle-o fa-5x" />
                </div>
                <div
                  className="col-xs-9 text-right">
                  <div
                    className="huge"
                    style={
                      Object {
                        "fontSize": "40px",
                      }
                    }>
                    0
                  </div>
                  <div>
                    Active Users This Month
                  </div>
                </div>
              </div>
            </div>
          </div>
        </Component>
      </div>
    </div>
    <div
      className="row">
      <div
        className="MessageChart Chart col-lg-12 col-md-12">
        <h3
          className="text-center">
          Delivered, Read, Undelivered Messages Ratio
        </h3>
        <ResponsiveContainer
          debounce={0}
          height="80%"
          minHeight={360}
          style={
            Object {
              "margin": "auto",
            }
          }
          width="80%">
          <div
            className="recharts-responsive-container"
            style={
              Object {
                "height": "80%",
                "maxHeight": undefined,
                "minHeight": 360,
                "minWidth": undefined,
                "width": "80%",
              }
            }>
            <ComposedDataDecorator(PieChart)
              height={0}
              margin={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "right": 0,
                  "top": 0,
                }
              }
              style={Object {}}
              width={0}>
              <PieChart
                allComposedData={
                  Array [
                    Array [
                      Object {
                        "fill": "#31708f",
                        "name": "Delivered and Read",
                        "payload": Object {
                          "name": "Delivered and Read",
                          "value": 0,
                        },
                        "stroke": "#fff",
                        "value": 0,
                      },
                      Object {
                        "fill": "#31708f",
                        "name": "Delivered, but Not Read",
                        "payload": Object {
                          "name": "Delivered, but Not Read",
                          "value": 0,
                        },
                        "stroke": "#fff",
                        "value": 0,
                      },
                      Object {
                        "fill": "#31708f",
                        "name": "Not Delivered",
                        "payload": Object {
                          "name": "Not Delivered",
                          "value": 0,
                        },
                        "stroke": "#fff",
                        "value": 0,
                      },
                    ],
                  ]
                }
                height={0}
                margin={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "right": 0,
                    "top": 0,
                  }
                }
                style={Object {}}
                width={0} />
            </ComposedDataDecorator(PieChart)>
            <ResizeDetector
              handleHeight={true}
              handleWidth={true}
              onResize={[Function]}>
              <div
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "visibility": "hidden",
                    "zIndex": -1,
                  }
                }>
                <div
                  onScroll={[Function]}
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "visibility": "hidden",
                      "zIndex": -1,
                    }
                  }>
                  <div
                    style={
                      Object {
                        "height": 10,
                        "left": 0,
                        "position": "absolute",
                        "top": 0,
                        "width": 10,
                      }
                    } />
                </div>
                <div
                  onScroll={[Function]}
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "visibility": "hidden",
                      "zIndex": -1,
                    }
                  }>
                  <div
                    style={
                      Object {
                        "height": "200%",
                        "left": 0,
                        "position": "absolute",
                        "top": 0,
                        "width": "200%",
                      }
                    } />
                </div>
              </div>
            </ResizeDetector>
          </div>
        </ResponsiveContainer>
      </div>
    </div>
    <div
      className="row">
      <div
        className="col-lg-4 col-md-4" />
      <div
        className="col-lg-4 col-md-4">
        <div
          className="TotalErrors">
          <Component
            icon="fa-exclamation-triangle"
            label="No. of message errors"
            number={0}>
            <div
              className="panel panel-info">
              <div
                className="panel-heading">
                <div
                  className="row">
                  <div
                    className="col-xs-3">
                    <i
                      className="fa fa-exclamation-triangle fa-5x" />
                  </div>
                  <div
                    className="col-xs-9 text-right">
                    <div
                      className="huge"
                      style={
                        Object {
                          "fontSize": "40px",
                        }
                      }>
                      0
                    </div>
                    <div>
                      No. of message errors
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </Component>
        </div>
      </div>
      <div
        className="col-lg-4 col-md-4" />
    </div>
    <div
      className="row">
      <div
        className="col-lg-4 col-md-4">
        <Component
          icon="fa-comments"
          label="Total Messages"
          number={0}>
          <div
            className="panel panel-info">
            <div
              className="panel-heading">
              <div
                className="row">
                <div
                  className="col-xs-3">
                  <i
                    className="fa fa-comments fa-5x" />
                </div>
                <div
                  className="col-xs-9 text-right">
                  <div
                    className="huge"
                    style={
                      Object {
                        "fontSize": "40px",
                      }
                    }>
                    0
                  </div>
                  <div>
                    Total Messages
                  </div>
                </div>
              </div>
            </div>
          </div>
        </Component>
      </div>
      <div
        className="col-lg-4 col-md-4">
        <Component
          icon="fa-comment"
          label="Avg. no. messages in a conversation"
          number="0.00">
          <div
            className="panel panel-info">
            <div
              className="panel-heading">
              <div
                className="row">
                <div
                  className="col-xs-3">
                  <i
                    className="fa fa-comment fa-5x" />
                </div>
                <div
                  className="col-xs-9 text-right">
                  <div
                    className="huge"
                    style={
                      Object {
                        "fontSize": "40px",
                      }
                    }>
                    0.00
                  </div>
                  <div>
                    Avg. no. messages in a conversation
                  </div>
                </div>
              </div>
            </div>
          </div>
        </Component>
      </div>
      <div
        className="col-lg-4 col-md-4">
        <Component
          icon="fa-commenting"
          label="Avg. length of a conversation"
          number="">
          <div
            className="panel panel-info">
            <div
              className="panel-heading">
              <div
                className="row">
                <div
                  className="col-xs-3">
                  <i
                    className="fa fa-commenting fa-5x" />
                </div>
                <div
                  className="col-xs-9 text-right">
                  <div
                    className="huge"
                    style={
                      Object {
                        "fontSize": "40px",
                      }
                    } />
                  <div>
                    Avg. length of a conversation
                  </div>
                </div>
              </div>
            </div>
          </div>
        </Component>
      </div>
    </div>
  </div>
</Dashboard>
`;

exports[`Dashboard renders correctly 2`] = `
<Dashboard>
  <div
    className="container-fluid">
    <h1>
      Dashboard
    </h1>
    <div
      className="row Dashboard">
      <div
        className="UserChart Chart">
        <h3
          className="text-center">
          Active Users By Month
        </h3>
        <ResponsiveContainer
          debounce={0}
          height="80%"
          minHeight={360}
          style={
            Object {
              "margin": "auto",
            }
          }
          width="80%">
          <div
            className="recharts-responsive-container"
            style={
              Object {
                "height": "80%",
                "maxHeight": undefined,
                "minHeight": 360,
                "minWidth": undefined,
                "width": "80%",
              }
            }>
            <AnimationDecorator(ComposedDataDecorator(BarChart))
              data={
                Array [
                  Object {
                    "count": 1,
                    "month": "April",
                  },
                ]
              }
              height={0}
              layout="horizontal"
              margin={
                Object {
                  "bottom": 5,
                  "left": 5,
                  "right": 5,
                  "top": 5,
                }
              }
              stackOffset="none"
              width={0} />
            <ResizeDetector
              handleHeight={true}
              handleWidth={true}
              onResize={[Function]}>
              <div
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "visibility": "hidden",
                    "zIndex": -1,
                  }
                }>
                <div
                  onScroll={[Function]}
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "visibility": "hidden",
                      "zIndex": -1,
                    }
                  }>
                  <div
                    style={
                      Object {
                        "height": 10,
                        "left": 0,
                        "position": "absolute",
                        "top": 0,
                        "width": 10,
                      }
                    } />
                </div>
                <div
                  onScroll={[Function]}
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "visibility": "hidden",
                      "zIndex": -1,
                    }
                  }>
                  <div
                    style={
                      Object {
                        "height": "200%",
                        "left": 0,
                        "position": "absolute",
                        "top": 0,
                        "width": "200%",
                      }
                    } />
                </div>
              </div>
            </ResizeDetector>
          </div>
        </ResponsiveContainer>
      </div>
    </div>
    <div
      className="row">
      <div
        className="col-lg-4 col-md-4">
        <Component
          icon="fa-users"
          label="Total Users"
          number={1}>
          <div
            className="panel panel-info">
            <div
              className="panel-heading">
              <div
                className="row">
                <div
                  className="col-xs-3">
                  <i
                    className="fa fa-users fa-5x" />
                </div>
                <div
                  className="col-xs-9 text-right">
                  <div
                    className="huge"
                    style={
                      Object {
                        "fontSize": "40px",
                      }
                    }>
                    1
                  </div>
                  <div>
                    Total Users
                  </div>
                </div>
              </div>
            </div>
          </div>
        </Component>
      </div>
      <div
        className="col-lg-4 col-md-4">
        <Component
          icon="fa-user"
          label="Active Users Today"
          number={0}
          previous={0}>
          <div
            className="panel panel-warning">
            <div
              className="panel-heading">
              <div
                className="row">
                <div
                  className="col-xs-3">
                  <i
                    className="fa fa-user fa-5x" />
                </div>
                <div
                  className="col-xs-9 text-right">
                  <div
                    className="huge"
                    style={
                      Object {
                        "fontSize": "40px",
                      }
                    }>
                    0
                  </div>
                  <div>
                    Active Users Today
                  </div>
                </div>
              </div>
            </div>
          </div>
        </Component>
      </div>
      <div
        className="col-lg-4 col-md-4">
        <Component
          icon="fa-user-circle-o"
          label="Active Users This Month"
          number={1}
          previous={0}>
          <div
            className="panel panel-success">
            <div
              className="panel-heading">
              <div
                className="row">
                <div
                  className="col-xs-3">
                  <i
                    className="fa fa-user-circle-o fa-5x" />
                </div>
                <div
                  className="col-xs-9 text-right">
                  <div
                    className="huge"
                    style={
                      Object {
                        "fontSize": "40px",
                      }
                    }>
                    1
                  </div>
                  <div>
                    Active Users This Month
                  </div>
                </div>
              </div>
            </div>
          </div>
        </Component>
      </div>
    </div>
    <div
      className="row">
      <div
        className="MessageChart Chart col-lg-12 col-md-12">
        <h3
          className="text-center">
          Delivered, Read, Undelivered Messages Ratio
        </h3>
        <ResponsiveContainer
          debounce={0}
          height="80%"
          minHeight={360}
          style={
            Object {
              "margin": "auto",
            }
          }
          width="80%">
          <div
            className="recharts-responsive-container"
            style={
              Object {
                "height": "80%",
                "maxHeight": undefined,
                "minHeight": 360,
                "minWidth": undefined,
                "width": "80%",
              }
            }>
            <ComposedDataDecorator(PieChart)
              height={0}
              margin={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "right": 0,
                  "top": 0,
                }
              }
              style={Object {}}
              width={0}>
              <PieChart
                allComposedData={
                  Array [
                    Array [
                      Object {
                        "fill": "#31708f",
                        "name": "Delivered and Read",
                        "payload": Object {
                          "name": "Delivered and Read",
                          "value": 16,
                        },
                        "stroke": "#fff",
                        "value": 16,
                      },
                      Object {
                        "fill": "#31708f",
                        "name": "Delivered, but Not Read",
                        "payload": Object {
                          "name": "Delivered, but Not Read",
                          "value": 0,
                        },
                        "stroke": "#fff",
                        "value": 0,
                      },
                      Object {
                        "fill": "#31708f",
                        "name": "Not Delivered",
                        "payload": Object {
                          "name": "Not Delivered",
                          "value": 1,
                        },
                        "stroke": "#fff",
                        "value": 1,
                      },
                    ],
                  ]
                }
                height={0}
                margin={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "right": 0,
                    "top": 0,
                  }
                }
                style={Object {}}
                width={0} />
            </ComposedDataDecorator(PieChart)>
            <ResizeDetector
              handleHeight={true}
              handleWidth={true}
              onResize={[Function]}>
              <div
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "visibility": "hidden",
                    "zIndex": -1,
                  }
                }>
                <div
                  onScroll={[Function]}
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "visibility": "hidden",
                      "zIndex": -1,
                    }
                  }>
                  <div
                    style={
                      Object {
                        "height": 10,
                        "left": 0,
                        "position": "absolute",
                        "top": 0,
                        "width": 10,
                      }
                    } />
                </div>
                <div
                  onScroll={[Function]}
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "visibility": "hidden",
                      "zIndex": -1,
                    }
                  }>
                  <div
                    style={
                      Object {
                        "height": "200%",
                        "left": 0,
                        "position": "absolute",
                        "top": 0,
                        "width": "200%",
                      }
                    } />
                </div>
              </div>
            </ResizeDetector>
          </div>
        </ResponsiveContainer>
      </div>
    </div>
    <div
      className="row">
      <div
        className="col-lg-4 col-md-4" />
      <div
        className="col-lg-4 col-md-4">
        <div
          className="TotalErrors">
          <Component
            icon="fa-exclamation-triangle"
            label="No. of message errors"
            number={0}>
            <div
              className="panel panel-info">
              <div
                className="panel-heading">
                <div
                  className="row">
                  <div
                    className="col-xs-3">
                    <i
                      className="fa fa-exclamation-triangle fa-5x" />
                  </div>
                  <div
                    className="col-xs-9 text-right">
                    <div
                      className="huge"
                      style={
                        Object {
                          "fontSize": "40px",
                        }
                      }>
                      0
                    </div>
                    <div>
                      No. of message errors
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </Component>
        </div>
      </div>
      <div
        className="col-lg-4 col-md-4" />
    </div>
    <div
      className="row">
      <div
        className="col-lg-4 col-md-4">
        <Component
          icon="fa-comments"
          label="Total Messages"
          number={34}>
          <div
            className="panel panel-info">
            <div
              className="panel-heading">
              <div
                className="row">
                <div
                  className="col-xs-3">
                  <i
                    className="fa fa-comments fa-5x" />
                </div>
                <div
                  className="col-xs-9 text-right">
                  <div
                    className="huge"
                    style={
                      Object {
                        "fontSize": "40px",
                      }
                    }>
                    34
                  </div>
                  <div>
                    Total Messages
                  </div>
                </div>
              </div>
            </div>
          </div>
        </Component>
      </div>
      <div
        className="col-lg-4 col-md-4">
        <Component
          icon="fa-comment"
          label="Avg. no. messages in a conversation"
          number="5.67">
          <div
            className="panel panel-info">
            <div
              className="panel-heading">
              <div
                className="row">
                <div
                  className="col-xs-3">
                  <i
                    className="fa fa-comment fa-5x" />
                </div>
                <div
                  className="col-xs-9 text-right">
                  <div
                    className="huge"
                    style={
                      Object {
                        "fontSize": "40px",
                      }
                    }>
                    5.67
                  </div>
                  <div>
                    Avg. no. messages in a conversation
                  </div>
                </div>
              </div>
            </div>
          </div>
        </Component>
      </div>
      <div
        className="col-lg-4 col-md-4">
        <Component
          icon="fa-commenting"
          label="Avg. length of a conversation"
          number="00:08">
          <div
            className="panel panel-info">
            <div
              className="panel-heading">
              <div
                className="row">
                <div
                  className="col-xs-3">
                  <i
                    className="fa fa-commenting fa-5x" />
                </div>
                <div
                  className="col-xs-9 text-right">
                  <div
                    className="huge"
                    style={
                      Object {
                        "fontSize": "40px",
                      }
                    }>
                    00:08
                  </div>
                  <div>
                    Avg. length of a conversation
                  </div>
                </div>
              </div>
            </div>
          </div>
        </Component>
      </div>
    </div>
  </div>
</Dashboard>
`;
